src/main/scala/com.snowplowanalytics.sauna/actors/Mediator.scala

```diff
@@ -390,6 +398,60 @@ object Mediator {
   }
 
   /**
+   * A function producing `Props` based on loggers for the Hipchat responder.
+   *
+   * @param saunaSettings A global settings object.
+   * @return A list of functions that accept loggers and produce Hipchat responders.
+   */
+  def hipchatCreator(saunaSettings: SaunaSettings): List[ActorConstructor] = {
+    saunaSettings.hipchatResponderConfig match {
+      case Some(responders.HipchatConfig(true, id, params)) =>
+        val apiWrapper: SaunaLogger => Hipchat = (logger) => new Hipchat(params.authToken, logger)
+        if (params.sendRoomNotificationEnabled) {
+          ((logger: SaunaLogger) => (id, SendRoomNotificationResponder.props(apiWrapper(logger), logger))) :: Nil
+        } else Nil
+
+      case _ => Nil
+    }
+  }
+
+  /**
+   * A function producing `Props` based on loggers for the Slack responder.
+   *
+   * @param saunaSettings A global settings object.
+   * @return A list of functions that accept loggers and produce Slack responders.
+   */
+  def slackCreator(saunaSettings: SaunaSettings): List[ActorConstructor] = {
+    saunaSettings.slackConfig match {
+      case Some(responders.SlackConfig(true, id, params)) =>
+        val apiWrapper: SaunaLogger => Slack = (logger) => new Slack(params.webhookUrl, logger)
+        if (params.sendMessageEnabled) {
+          ((logger: SaunaLogger) => (id, SendMessageResponder.props(apiWrapper(logger), logger))) :: Nil
+        } else Nil
+
+      case _ => Nil
+    }
+  }
+
+  /**
+   * A function producing `Props` based on loggers for the PagerDuty responder.
+   *
+   * @param saunaSettings A global settings object.
+   * @return A list of functions that accept loggers and produce PagerDuty responders.
+   */
+  def pagerDutyCreator(saunaSettings: SaunaSettings): List[ActorConstructor] = {
+    saunaSettings.pagerDutyConfig match {
+      case Some(responders.PagerDutyConfig(true, id, params)) =>
+        val apiWrapper: SaunaLogger => PagerDuty = (logger) => new PagerDuty(logger)
+        if (params.createEventEnabled) {
+          ((logger: SaunaLogger) => (id, CreateEventResponder.props(apiWrapper(logger), logger))) :: Nil
+        } else Nil
+
+      case _ => Nil
+    }
+  }
```

Comment:
Most of the `xCreator()` methods will eventually return multiple `ActorConstructor`s (e.g. https://github.com/snowplow/sauna/blob/hipchat-responder/src/main/scala/com.snowplowanalytics.sauna/actors/Mediator.scala#L357-L379) with distinct constructor signatures - I don't think there's a clean way to generalize that into a helper method.